models/stg_google_play__store_performance_source.sql

```diff
@@ -0,0 +1,40 @@
+
+with base as (
+
+    select * 
+    from {{ ref('stg_google_play__store_performance_source_tmp') }}
+
+),
+
+fields as (
+
+    select
+        {{
+            fivetran_utils.fill_staging_columns(
+                source_columns=adapter.get_columns_in_relation(ref('stg_google_play__store_performance_source_tmp')),
+                staging_columns=get_stats_store_performance_traffic_source_columns()
+            )
+        }}
+        
+    from base
+),
+
+final as (
+    
+    select 
+        date as date_day,
+        package_name,
+        traffic_source,
+        search_term,
+        utm_campaign,
+        utm_source,
+        store_listing_acquisitions,
+        store_listing_conversion_rate,
+        store_listing_visitors,
+        -- make a surrogate key as the PK involves quite a few columns
+        {{ dbt_utils.surrogate_key(['date', 'package_name', 'traffic_source', 'search_term', 'utm_campaign', 'utm_source']) }} as traffic_source_unique_key,
```

Comment:
Out of curiosity, is this method comparable to [unique_combination_of_columns](https://github.com/dbt-labs/dbt-utils#unique_combination_of_columns-source)? When is one preferred over the other?

Reply:
I'd also like to hear your thoughts on decluttering the actual code here and moving this comment to a `DECISIONLOG`?

Reply:
yah so i decided to make a surrogate key instead to provide an easy-to-use primary key for this table, as its grain (a composite of 6 columns) is kinda confusing and not aligned with the other tables. 

idk if this is a big enough decision to merit a decision log entry, and the comment may just be more digestable to package users, but very open to other opinions

Reply:
Sounds good!

Reply:
Sounds good!